src/core/reducers/addons.js

```diff
@@ -1,15 +1,36 @@
 /* @flow */
 import { ADDON_TYPE_THEME } from 'core/constants';
 import type { ErrorHandlerType } from 'core/errorHandler';
-import type { AddonType } from 'core/types/addons';
+import type { AddonType, ExternalAddonType } from 'core/types/addons';
 
 
+export const ADDONS_LOADED = 'ADDONS_LOADED';
 export const FETCH_ADDON = 'FETCH_ADDON';
 
-const initialState = {};
+type ExternalAddonMap = {
+  [addonSlug: string]: ExternalAddonType,
+};
+
+export type LoadAddonsAction = {|
+  payload: {| addons: ExternalAddonMap |},
+  type: string,
+|};
 
-type Action = Object;
-type AddonState = Object;
+export function loadAddons(
+  entities: {| addons?: ExternalAddonMap |}
```

Comment:
As mentioned previously I think it would be great to keep flow definitions out of the function signatures if they're being included. They really crowd the function signature and make it very difficult to parse exactly what's happening inside it.

Reply:
The issue I have with making them separate is that it slows down the reader. If you want to know what the type is, you have to cross-reference it from somewhere else. I understand how separation can be helpful for big definitions but this one only spans about 30 characters so I don't think that warrants separation.